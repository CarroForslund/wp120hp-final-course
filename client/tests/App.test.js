import { shallow, render } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import Layout from "../pages/index.js";
import Grid from "../pages/index.js";
import Card from "../components/card.js";
import AddTranslation from "../pages/translations/add.js";

describe("With Snapshot Testing", () => {
  test('App shows "Welcome to Equi Translation App"', () => {
    const component = renderer.create(<Layout />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Testing with Enzyme", () => {
  test('App shows "Welcome to Equi Translation App" H1', () => {
    const app = render(<Layout />);
    expect(app.find("h2").text()).toEqual("Welcome to Equi Translation App");
  });

  // Link to all translations is rendered
  test('App shows "View translations" link H3', () => {
    const app = shallow(<Grid />);
    expect(app.find(Card).at(0).render().find("h3").text()).toEqual("View translations →");
  });

  // Link to add translation is rendered
  test('App shows "Add translation" link H3', () => {
    const app = shallow(<Grid />);
    expect(app.find(Card).at(1).render().find("h3").text()).toEqual("Add new →");
  });

  // Add new translation page is rendered
  test('Page to add translation shows "Add Translation" H1', () => {
    const app = render(<AddTranslation />);
    expect(app.find("h1").text()).toEqual("Add Translation");
  });

  
});

