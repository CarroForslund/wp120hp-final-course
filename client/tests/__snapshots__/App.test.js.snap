// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "Welcome to Equi Translation App" 1`] = `
Array [
  <div>
    <header>
      <img
        alt="Equi Translation App"
        className="undefined undefined"
        src="/images/head-icon-6529.png"
      />
      <h1>
        Equi Translation App
      </h1>
    </header>
    <main>
      <h2>
        Welcome to Equi Translation App
      </h2>
      <p>
        Here can you look for existing Swedish/English translations of horse related words and phrases, and add missing ones to the library. Note that this app is still under development.
      </p>
      <div>
        <a
          href="/translations"
        >
          <h3>
            View translations →
          </h3>
          <p>
            See a list of all the words added to the library.
            .
          </p>
        </a>
        <a
          href="/translations/add"
        >
          <h3>
            Add new →
          </h3>
          <p>
            Here you can add a new translation to the library.
            .
          </p>
        </a>
      </div>
    </main>
  </div>,
  <footer>
    <a
      href="https://codingcaroline.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      Powered by Coding Caroline
    </a>
  </footer>,
]
`;
